### DISCUSSION TITLE
RTK Query isFetching reason

### DISCUSSION THREAD

**Discussion by aphilas** [reduxjs/redux-toolkit#3884] [Q&A] [answered]
I want to show a loading spinner only for the initial load, and user initiated actions.

The alternatives are:
* use `isLoading` only - when a user hits next, or enters a search query they have no idea something is loading in the background
* use `isLoading || isFetching`
  * the user will see _spurious_ fetches, since I have `refetchOnFocus`
  * the current UI is of the form `isLoading ? <Loading /> : <Data />`. Thus with `isLoading || isFetching` the UI will keep "reflowing" every time (on focus, on adding a single character to the query [even with debounce, if you type slowly enough]). 

Is it possible to get a reason for why the refetch is happening .e.g. `ParamsChanged | FocusReceived | InvalidationTimeout`...

While I have searched the issues/discussions/docs, I might have missed sth trivial...

**Comment by phryneas** [comment, marked as answer]
No, there's nothing like that, sorry.

**Reply by aphilas** [reply]
Oops.

Thanks.
